### PR TITLE
Update EEPs have more consistent header contents

### PR DESCRIPTION
**Pull request by garazdawi**
As the headers in the EEPs are mostly freetext, the contents have diverged depending on author. This PR brings back some consistency and adds some checks so that it will not diverge more in the future.

The checks added are;

1. Erlang-Version must be in the format of an Erlang/OTP version tag and must be present if an EEP is in Final state.
2. Github actions actually check that the current checks are followed.

### CODE REVIEW

File: eeps/eep-0001.md [hunk 1]
```diff
@@ -272,8 +272,9 @@ versions of the EEP are posted to erlang-questions.  Both headers
 should be in dd-mmm-yyyy format, e.g. 14-Aug-2009.
 
 Standards Track EEPs must have a Erlang-Version header which indicates
-the version of Erlang that the feature will be released with.  Process
-EEPs do not need a Erlang-Version header.
+the version of Erlang that the feature will or has be released with.
```
Comment (kikofernandez) [inline eeps/eep-0001.md:275]:
```suggestion
the version of Erlang that the feature will or has been released with.  
```

File: eeps/eep-0011.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
      except for Unicode support according to EEP 10
     Type: Standards Track
     Created: 04-Jun-2008
-    Erlang-Version: R12B-5
+    Erlang-Version: OTP_R12B-5
```
Comment (kikofernandez) [inline eeps/eep-0011.md:6]: Do we need to change then the `Status` as well?
Currently, it says:
```
Status: Accepted/R12B-3u Proposal is implemented in OTP release R12B-3,
```
but we just wrote the the version is written `OTP_R12B-X`

File: eeps/eep-0016.md [hunk 3]
```diff
@@ -181,12 +181,13 @@ it would be easier to add this than to write it up.
 
 Here's some text to go in the documentation:
 
+<!-- markdownlint-disable MD027 -->
```
Comment (kikofernandez) [inline eeps/eep-0016.md:184]: Curious: what is this for?

File: eeps/eep-0044.md [hunk 4]
```diff
@@ -3,7 +3,7 @@
             are implemented in OTP version 19.0
```
Comment (kikofernandez) [inline eeps/eep-0044.md:3]: in `eep-0045.md` the text says:

```
Status: Accepted/19.0  Implemented in OTP release 19
```

In this file `eep-0044.md` we say:

```
Status: Accepted/19.0-we  Proposal's -warning and -error directives
            are implemented in OTP version 19.0
```

We need consistency :) either 19.0 or 19.

File: eeps/eep-0045.md [hunk 5]
```diff
@@ -1,8 +1,8 @@
     Author: Björn Gustavsson <bjorn(at)erlang(dot)org>
-    Status: Accepted/19.0  Implemented in OTP version 19.0
+    Status: Accepted/19.0  Implemented in OTP release 19
```
Comment (kikofernandez) [inline eeps/eep-0045.md:2]: Here is the mention of 19.0 and release 19.
When is ok to say 19.0 and when is ok to simply say release 19?
Maybe this is something that should be in the description, so that we make some guidelines?

File: eeps/eep-0043.md [hunk 6]
```diff
@@ -2,7 +2,7 @@
     Status:         Final/R17B  Implemented in OTP R17B (except ...)
     Type:           Standards Track
     Created:        04-Apr-2013
-    Erlang-Version: R17A
+    Erlang-Version: OTP_R17A
```
Comment (kikofernandez) [inline eeps/eep-0043.md:5]: We were using `OTP-RXXX` and then we switch to `OTP-19.0`
Is this change intended? Why not make everything `OTP-RXXX` and `OTP-XXX`.
Just curious to understand the change.

File: eeps/eep-0058.md [hunk 7]
```diff
@@ -1,8 +1,8 @@
     Author: Sergey Prokhorov <seriy(dot)pr(at)gmail(dot)com>
-    Status: Final/26.0 Implemented in OTP release 26.0
+    Status: Final/26.0 Implemented in OTP release 26
     Type: Standards Track
     Created: 14-Sep-2021
-    Erlang-Version: 25
+    Erlang-Version: OTP-26.0
```
Comment (kikofernandez) [inline eeps/eep-0058.md:5]: I guess this was a typo in the EEP?

File: eeps/eep-0069.md [hunk 8]
```diff
@@ -1,4 +1,4 @@
-    Author: Isabell Huang <isabell.huang@ericsson.com>
+    Author: Isabell Huang <isabell(at)erlang(dot)org>
     Status: Accepted
     Type: Standards Track
```
Comment (kikofernandez) [inline eeps/eep-0069.md:3]: I guess we should add the Erlang-Version when this gets merged in OTP-28?

File: eeps/eep-0043.md (hunk 6 above)
Comment (garazdawi) [reply eeps/eep-0043.md:5]:
```suggestion
    Erlang-Version: OTP-17.0
```

Releases before 17 were created using tags that look like this: `OTP_RXBY` and then with 17 we changed to `OTP-X.Y.Z` to be more compatible with how other opensource projects use versions.

File: eeps/eep-0044.md (hunk 4 above)
Comment (garazdawi) [reply eeps/eep-0044.md:3]: It should be updated to `release 19`, I missed this one.

File: eeps/eep-0058.md (hunk 7 above)
Comment (garazdawi) [reply eeps/eep-0058.md:5]: No, not really. It was the Erlang version that the EEP was originally target at, but then it was released in a later version.

File: eeps/eep-0045.md (hunk 5 above)
Comment (garazdawi) [reply eeps/eep-0045.md:2]: 19.0 is a version and 19 is a release. So depending on what you talk about to use either or.

File: eeps/eep-0069.md (hunk 8 above)
Comment (garazdawi) [reply eeps/eep-0069.md:3]: yes, when changing the status to final.

**Review by kikofernandez — APPROVED** [review]
